src/SIL.XForge.Scripture/Services/SFProjectService.cs

```diff
@@ -571,71 +602,86 @@ public async Task<IReadOnlyList<InviteeStatus>> InvitedUsersAsync(string curUser
     }
 
     /// <summary> Check that a share link is valid for a project and add the user to the project. </summary>
-    public async Task CheckLinkSharingAsync(string curUserId, string projectId, string shareKey)
+    public async Task<string> CheckLinkSharingAsync(string curUserId, string shareKey)
     {
         await using IConnection conn = await RealtimeService.ConnectAsync(curUserId);
+        ProjectSecret projectSecret = ProjectSecrets
+            .Query()
+            .FirstOrDefault(ps => ps.ShareKeys.Any(sk => sk.Key == shareKey));
+        if (projectSecret == null)
+            throw new ForbiddenException();
+
+        String projectId = projectSecret.Id;
+        ShareKey projectSecretShareKey = projectSecret.ShareKeys.FirstOrDefault(sk => sk.Key == shareKey);
+
+        if (projectSecretShareKey.RecipientUserId != null)
```

Comment:
We should change this line to `if (projectSecretShareKey?.RecipientUserId != null)`, as `projectSecretShareKey` can be null.